From 839673b8f5bd0d49d2b6e2ae0f2c2b15e53a138d Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Tue, 14 Aug 2012 13:50:37 +0530
Subject: [PATCH 6/6] pwm: ehrpwm: Configure polarity on pwm_start

Polarity of PWM device should done only from start. This helps in
disabling polarity change even PWM device is running.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 drivers/pwm/ehrpwm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pwm/ehrpwm.c b/drivers/pwm/ehrpwm.c
index 8bbed87..636b0d9 100644
--- a/drivers/pwm/ehrpwm.c
+++ b/drivers/pwm/ehrpwm.c
@@ -1284,8 +1284,8 @@ static int ehrpwm_pwm_config(struct pwm_device *p,
 		break;
 
 	case BIT(PWM_CONFIG_POLARITY):
+		/* Polarity configuration done from pwm_start */
 		p->active_high = c->polarity;
-		ret = ehrpwm_pwm_set_pol(p);
 		break;
 
 	case BIT(PWM_CONFIG_START):
-- 
1.8.1.4

